pkg/preflights/host-preflight.yaml

```
@@ -370,6 +424,342 @@ spec:
+        checkName: ETCD Internal Port Availability
+        collectorName: ETCD Internal Port
+        outcomes:
+          - fail:
+              when: "connection-refused"
+              message: Connection to port 2379/TCP was refused. This may indicate that the connection is being redirected to somewhere else on your network (perhaps a proxy configuration).
+          - fail:
+              when: "address-in-use"
+              message: Port 2379/TCP is required, but another process is already listening on it. Relocate the conflicting process to continue.
+          - fail:
+              when: "connection-timeout"
+              message: Port 2379/TCP is required, but the connection timed out. Ensure that your firewall doesn't block port 2379/TCP.
+          - fail:
+              when: "error"
+              message: Port 2379/TCP is required, but an unexpected error occurred when trying to connect to it
+          - pass:
+              when: "connected"
+              message: Port 2379/TCP is available
+          - error:
+              message: Port 2379/TCP is required, but an unexpected error occurred when trying to connect to it
+    - tcpPortStatus:
+        checkName: ETCD External Port Availability
+        collectorName: ETCD External Port
+        outcomes:
+          - fail:
+              when: "connection-refused"
+              message: Port 2380/TCP is required, but the connection was refused. This may indicate that the connection is being redirected to somewhere else on your network (perhaps a proxy configuration).
+          - fail:
+              when: "address-in-use"
+              message: Port 2380/TCP is required, but another process is already listening on it. Relocate the conflicting process to continue.
+          - fail:
+              when: "connection-timeout"
+              message: Port 2380/TCP is required, but the connection timed out. Ensure that your firewall doesn't block port 2380/TCP.
+          - fail:
+              when: "error"
+              message: Port 2380/TCP is required, but an unexpected error occurred when trying to connect to it
+          - pass:
+              when: "connected"
+              message: Port 2380/TCP is available
+          - error:
+              message: Port 2380/TCP is required, but an unexpected error occurred when trying to connect to it
+    - tcpPortStatus:
+        checkName: Local Artifact Mirror Port Availability
+        collectorName: Local Artifact Mirror Port
+        outcomes:
+          - fail:
+              when: "connection-refused"
+              message: Port 50000/TCP is required, but the connection was refused. This may indicate that the connection is being redirected to somewhere else on your network (perhaps a proxy configuration).
+          - fail:
+              when: "address-in-use"
+              message: Port 50000/TCP is required, but another process is already listening on it. Relocate the conflicting process to continue.
```

When we merge Ethan's port relocation changes, we can update this:
```suggestion
              message: Port 50000/TCP is required, but another process is already listening on it. To proceed, relocate the conflicting process or specify an available port using --local-artifact-mirror-port.
```